File: nexora/scripts/smoke.py
from pathlib import Path
from playwright.sync_api import sync_playwright

BASE = "http://127.0.0.1:5179"
OUT = Path("dist")
OUT.mkdir(exist_ok=True)
errors: list[str] = []


def visit(page, path: str, expect: str | None = None) -> None:
    res = page.goto(BASE + path, wait_until="domcontentloaded")
    if res is None or res.status >= 400:
        errors.append(f"{path} status {getattr(res, 'status', None)}")
    if expect:
        loc = page.get_by_text(expect).first
        if not loc.is_visible():
            errors.append(f'{path} missing "{expect}"')


def main() -> None:
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page(viewport={"width": 1440, "height": 900})
        page.on("pageerror", lambda err: errors.append(f"pageerror: {err}"))

        visit(page, "/", "Are you 18 or over")
        page.get_by_role("button", name="I am 18+").click()
        page.get_by_role("heading", name="The sharpest line in sport.").wait_for()
        page.screenshot(path=str(OUT / "smoke-home.png"))

        page.locator("button.odds").first.click()
        page.get_by_text("Bet slip").wait_for()
        page.screenshot(path=str(OUT / "smoke-slip.png"))

        for path, text in [
            ("/live", "Live betting"),
            ("/sports", "All sports"),
            ("/sports/football", "Football"),
            ("/event/epl_ars_liv", "Match Winner"),
            ("/casino", "Casino. Visual only"),
            ("/promotions", "Promotions"),
            ("/results", "Scores & settlement"),
            ("/leaderboards", "Leaderboards"),
            ("/help", "Help Centre"),
            ("/responsible-gambling", "Responsible gambling"),
            ("/about", "About Nexora"),
        ]:
            visit(page, path, text)

        visit(page, "/login", "Log in")
        page.get_by_role("button", name="Continue").click()
        page.get_by_text("Two-factor").wait_for()
        page.get_by_role("button", name="Verify").click()
        page.get_by_text("Hello, Alex").wait_for()
        page.screenshot(path=str(OUT / "smoke-account.png"))

        visit(page, "/wallet", "Balances")
        page.get_by_role("button", name="Deposit").click()
        page.get_by_role("button", name="Credit wallet").click()
        page.get_by_text("Deposit complete").first.wait_for(timeout=5000)

        page.goto(BASE + "/admin", wait_until="domcontentloaded")
        page.get_by_text("Admin only").wait_for()

        page.goto(BASE + "/account", wait_until="domcontentloaded")
        page.get_by_role("button", name="Log out").click()

        visit(page, "/login", "Log in")
        page.locator('input[type="email"]').fill("admin@nexora.demo")
        page.locator('input[type="password"]').fill("admin1234")
        page.get_by_role("button", name="Continue").click()
        page.get_by_role("button", name="Verify").click()
        page.goto(BASE + "/admin", wait_until="domcontentloaded")
        page.get_by_text("Operations").wait_for()
        page.screenshot(path=str(OUT / "smoke-admin.png"))

        mobile = browser.new_page(viewport={"width": 375, "height": 812})
        mobile.on("pageerror", lambda err: errors.append(f"mobile pageerror: {err}"))
        mobile.goto(BASE + "/", wait_until="domcontentloaded")
        gate = mobile.get_by_role("button", name="I am 18+")
        if gate.is_visible():
            gate.click()
        mobile.get_by_text("Home").first.wait_for()
        mobile.screenshot(path=str(OUT / "smoke-mobile.png"))
        mobile.get_by_role("button", name="Slip").click()
        mobile.get_by_text("Bet slip").wait_for()
        mobile.close()
        browser.close()

    if errors:
        print("SMOKE FAIL")
        for e in errors:
            print(" -", e)
        raise SystemExit(1)
    print("SMOKE OK")


if __name__ == "__main__":
    main()

File: nexora/scripts/smoke.mjs
import { chromium } from 'playwright';

const base = process.env.BASE_URL || 'http://127.0.0.1:5179';

const browser = await chromium.launch({ headless: true });
const errors = [];

async function shot(page, name) {
  await page.screenshot({ path: `dist/smoke-${name}.png`, fullPage: false });
}

async function visit(page, path, expectText) {
  const res = await page.goto(base + path, { waitUntil: 'domcontentloaded' });
  if (!res || res.status() >= 400) errors.push(`${path} status ${res?.status()}`);
  if (expectText) {
    const found = await page.getByText(expectText).first().isVisible().catch(() => false);
    if (!found) errors.push(`${path} missing "${expectText}"`);
  }
}

try {
  const page = await browser.newPage({ viewport: { width: 1440, height: 900 } });
  page.on('pageerror', (err) => errors.push('pageerror: ' + err.message));

  await visit(page, '/', 'Are you 18 or over');
  await page.getByRole('button', { name: 'I am 18+' }).click();
  await page.getByRole('heading', { name: /sharpest line/i }).waitFor();
  await shot(page, 'home');

  const odds = page.locator('button.odds').first();
  await odds.click();
  await page.getByText('Bet slip').waitFor();
  await shot(page, 'slip');

  for (const [path, text] of [
    ['/live', 'Live betting'],
    ['/sports', 'All sports'],
    ['/sports/football', 'Football'],
    ['/event/epl_ars_liv', 'Match Winner'],
    ['/casino', 'Casino. Visual only'],
    ['/promotions', 'Promotions'],
    ['/results', 'Scores & settlement'],
    ['/leaderboards', 'Leaderboards'],
    ['/help', 'Help Centre'],
    ['/responsible-gambling', 'Responsible gambling'],
    ['/about', 'About Nexora'],
  ]) {
    await visit(page, path, text);
  }

  await visit(page, '/login', 'Log in');
  await page.getByRole('button', { name: 'Continue' }).click();
  await page.getByText('Two-factor').waitFor();
  await page.getByRole('button', { name: 'Verify' }).click();
  await page.getByText('Hello, Alex').waitFor();
  await shot(page, 'account');

  await visit(page, '/wallet', 'Balances');
  await page.getByRole('button', { name: 'Deposit' }).click();
  await page.getByRole('button', { name: 'Credit wallet' }).click();
  await page.getByText(/Deposit complete|Balances/).first().waitFor();

  await visit(page, '/admin', 'Admin only');

  await page.getByRole('button', { name: 'Log out' }).click().catch(async () => {
    await page.goto(base + '/account');
  });

  await visit(page, '/login', 'Log in');
  await page.locator('input[type="email"]').fill('admin@nexora.demo');
  await page.locator('input[type="password"]').fill('admin1234');
  await page.getByRole('button', { name: 'Continue' }).click();
  await page.getByRole('button', { name: 'Verify' }).click();
  await page.goto(base + '/admin');
  await page.getByText('Operations').waitFor();
  await shot(page, 'admin');

  const mobile = await browser.newPage({ viewport: { width: 375, height: 812 } });
  mobile.on('pageerror', (err) => errors.push('mobile pageerror: ' + err.message));
  await mobile.goto(base + '/', { waitUntil: 'domcontentloaded' });
  const gate = mobile.getByRole('button', { name: 'I am 18+' });
  if (await gate.isVisible().catch(() => false)) await gate.click();
  await mobile.getByText('Home').first().waitFor();
  await shot(mobile, 'mobile');
  await mobile.getByRole('button', { name: /Slip/ }).click();
  await mobile.getByText('Bet slip').waitFor();
  await mobile.close();
} catch (err) {
  errors.push(String(err));
} finally {
  await browser.close();
}

if (errors.length) {
  console.error('SMOKE FAIL');
  for (const e of errors) console.error(' -', e);
  process.exit(1);
}
console.log('SMOKE OK');
